sansio_lsp_client/structs.py

```diff
@@ -203,19 +260,189 @@ class DiagnosticSeverity(enum.IntEnum):
     INFORMATION = 3
     HINT = 4
 
+    # TODO: delete this? it's easy to implement in your editor if you really need it
+    def short_name(self) -> str:
+        return {
+            self.ERROR: "Err",
+            self.WARNING: "Wrn",
+            self.INFORMATION: "Inf",
+            self.HINT: "Hint",
+        }[self]
+
 
 class Diagnostic(BaseModel):
     range: Range
 
-    # TODO: Make this a proper enum
-    # XXX: ^ Is this comment still relevant?
-    severity: DiagnosticSeverity
+    severity: t.Optional[DiagnosticSeverity]
 
-    # TODO: Support this as an union of str and int
-    code: t.Optional[t.Any]
+    code: t.Optional[t.Union[int, str]]
 
     source: t.Optional[str]
 
-    message: t.Optional[str]
+    message: str
 
     relatedInformation: t.Optional[t.List[DiagnosticRelatedInformation]]
+
+
+class MarkedString(BaseModel):
+    language: str
+    value: str
+
+
+class ParameterInformation(BaseModel):
+    label: t.Union[str, t.Tuple[int, int]]
+    documentation: t.Optional[t.Union[str, MarkupContent]]
+
+
+class SignatureInformation(BaseModel):
+    label: str
+    documentation: t.Optional[t.Union[MarkupContent, str]]
+    parameters: t.Optional[t.List[ParameterInformation]]
+    activeParameter: t.Optional[int]
+
+
+class SymbolKind(enum.IntEnum):
+    FILE = 1
+    MODULE = 2
+    NAMESPACE = 3
+    PACKAGE = 4
+    CLASS = 5
+    METHOD = 6
+    PROPERTY = 7
+    FIELD = 8
+    CONSTRUCTOR = 9
+    ENUM = 10
+    INTERFACE = 11
+    FUNCTION = 12
+    VARIABLE = 13
+    CONSTANT = 14
+    STRING = 15
+    NUMBER = 16
+    BOOLEAN = 17
+    ARRAY = 18
+    OBJECT = 19
+    KEY = 20
+    NULL = 21
+    ENUMMEMBER = 22
+    STRUCT = 23
+    EVENT = 24
+    OPERATOR = 25
+    TYPEPARAMETER = 26
+
+
+class SymbolTag(enum.IntEnum):
+    DEPRECATED = 1
+
+
+class CallHierarchyItem(BaseModel):
+    name: str
+    king: SymbolKind
+    tags: t.Optional[SymbolTag]
+    detail: t.Optional[str]
+    uri: str
+    range: Range
+    selectionRange: Range
+    data: t.Optional[t.Any]
+
+
+class CallHierarchyIncomingCall(BaseModel):
+    from_: CallHierarchyItem
+    fromRanges: t.List[Range]
+
+    class Config:
+        # 'from' is an invalid field - re-mapping
+        fields = {"from_": "from"}
+
+
+class CallHierarchyOutgoingCall(BaseModel):
+    to: CallHierarchyItem
+    fromRanges: t.List[Range]
+
+
+class TextDocumentSyncKind(enum.IntEnum):
+    NONE = 0
+    FULL = 1
+    INCREMENTAL = 2
+
+
+class SymbolInformation(BaseModel):  # symbols: flat list
+    name: str
+    kind: SymbolKind
+    tags: t.Optional[t.List[SymbolTag]]
+    deprecated: t.Optional[bool]
+    location: Location
+    containerName: t.Optional[str]
+
+
+class DocumentSymbol(BaseModel):  # symbols: hierarchy
```

Comment:
@halfbrained What do the `# symbols: ...` comments mean?

Reply:
Meaning is: `SymbolInformation` is just a flat list of symbols, `DocumentSymbol` - a tree - `METHOD` items are children of `CLASS` items, etc. 

It's not clear from just the type names.